e2e_playwright/st_main_layout_test.py

```diff
@@ -0,0 +1,336 @@
+):
+    """Test deploy button positioning on desktop after manually collapsing sidebar."""
+    setup_viewport_and_verify_title(app, 1280, 800, sidebar_mode)
+
+    # Manually collapse the sidebar
+    sidebar_header = app.get_by_test_id("stSidebarHeader")
+    sidebar_header.hover()
+
+    collapse_button = app.get_by_test_id("stSidebarCollapseButton").locator("button")
+    collapse_button.click()
+
+    # Verify sidebar is collapsed
+    verify_sidebar_state(app, False)
+
+    # Verify header adjusts correctly
+    header = app.get_by_test_id("stHeader")
+    expect(header).to_be_visible()
+
+    # Take screenshot focusing on header area
+    assert_snapshot(
+        header, name="st_main_layout-deploy_button_desktop_manually_collapsed"
+    )
+
+
+# Test responsive behavior during viewport size changes
+@pytest.mark.parametrize("sidebar_mode", ["auto"], indirect=True)
+def test_viewport_resize_responsive_behavior(
+    app: Page, assert_snapshot: ImageCompareFunction, sidebar_mode: str
+):
+    """Test sidebar behavior when resizing viewport with auto mode."""
+    # Start with desktop - sidebar should be expanded
+    setup_viewport_and_verify_title(app, 1280, 800, sidebar_mode)
+    verify_sidebar_state(app, True)
+
+    # Resize to mobile - sidebar should auto-collapse
+    app.set_viewport_size({"width": 375, "height": 667})
+
+    # Verify sidebar collapsed on mobile
+    verify_sidebar_state(app, False)
+    verify_expand_button_visible(app)
+
+    # Take final snapshot
+    assert_snapshot(app, name="st_main_layout-responsive_mobile_final")
+
+
+# Comprehensive responsiveness test
+@pytest.mark.parametrize(
+    ("sidebar_mode", "viewport_config"),
+    [
+        ("auto", {"name": "mobile", "width": 375, "height": 667}),
+        ("auto", {"name": "tablet", "width": 768, "height": 1024}),
+        ("auto", {"name": "desktop", "width": 1280, "height": 800}),
+        ("auto", {"name": "wide", "width": 1920, "height": 1080}),
+    ],
+    indirect=["sidebar_mode"],
+)
+def test_layout_responsiveness_auto_mode(
+    app: Page,
+    viewport_config: dict[str, Any],
+    assert_snapshot: ImageCompareFunction,
+    sidebar_mode: str,
+):
+    """Test layout responsiveness with auto sidebar mode across different viewport sizes."""
+    setup_viewport_and_verify_title(
+        app, viewport_config["width"], viewport_config["height"], sidebar_mode
+    )
+
+    # Verify basic layout elements are present
+    header = app.get_by_test_id("stHeader")
+    expect(header).to_be_visible()
+
+    main_content = app.get_by_test_id("stMain")
+    expect(main_content).to_be_visible()
+
+    sidebar = app.get_by_test_id("stSidebar")
+    expect(sidebar).to_be_attached()
+
+    expect(app.get_by_text("Info box 1")).to_be_visible()
+    expect(app.get_by_text("Info box 2")).to_be_visible()
+
+    # Take full app screenshot for this viewport
+    assert_snapshot(
+        app, name=f"st_main_layout-auto_{viewport_config['name']}_responsive"
```

Comment:
nit: the `{viewport_config['name']}` is probably not necessary since some unique info from every fixture is auto-included in the screenshot name.